src/backlight/backlight.c

```diff
@@ -386,17 +386,21 @@ int main(int argc, char *argv[]) {
                 if (!validate_device(udev, device))
                         return EXIT_SUCCESS;
 
-                r = read_one_line_file(saved, &value);
-                if (r < 0) {
-
-                        if (r == -ENOENT)
-                                return EXIT_SUCCESS;
+                clamp = udev_device_get_property_value(device, "ID_BACKLIGHT_CLAMP");
 
+                r = read_one_line_file(saved, &value);
+                if (r == -ENOENT && (!clamp || parse_boolean(clamp) != 0)) {
+                        /* not saved: fake "0" to trigger brightness clamping if necessary */
+                        value = strdup("0");
+                        if (!value) {
+                                log_oom();
+                                return EXIT_FAILURE;
+                        }
+                } else if (r < 0) {
                         log_error_errno(r, "Failed to read %s: %m", saved);
                         return EXIT_FAILURE;
                 }
 
-                clamp = udev_device_get_property_value(device, "ID_BACKLIGHT_CLAMP");
                 if (!clamp || parse_boolean(clamp) != 0) /* default to clamping */
```

Comment:
so now parse this twice. Let's clean this up properly then, let's add a new helper function:

```c
static bool shall_clamp(struct udev_device *d) {
        const char *s;
        int r;

        assert(d);

        s = udev_device_get_property_value(d, "ID_BACKLIGHT_CLAMP");
        if (!s)
                return true;

        r = parse_boolean(s);
        if (r < 0) {
                log_debug_errno(r, "Failed to parse ID_BACKLIGHT_CLAMP property, ignoring: %m");
                return true;
        }

        return r;
}
```

And then, let's change the `clamp` variable in the `main()` function to a bool, initialized it only once, and check it at the two places.

Reply:
Makes sense to me. I changed the patch accordingly.